cpp/src/unary/cast_ops.cu

```diff
@@ -308,6 +333,21 @@ struct dispatch_unary_cast_to {
                       output_mutable.begin<DeviceT>(),
                       fixed_point_unary_cast<SourceT, TargetT>{scale});
 
+    if constexpr (cudf::is_floating_point<SourceT>()) {
+      // For floating-point values, beside input nulls, we also need to set nulls for the output
+      // rows corresponding to NaN and inf in the input.
+      auto const d_input_ptr = column_device_view::create(input, stream);
+      auto [null_mask, null_count] =
+        cudf::detail::valid_if(thrust::make_counting_iterator(0),
```

Comment:
Don't we need to do an OR between the input's nulls and what this returns?  Or am I missing that somewhere.

Reply:
The input nulls are already considered in the functor of this.